builders.ncl

```diff
@@ -1,5 +1,12 @@
-let { NickelDerivation, Derivation, .. } = import "contracts.ncl" in
-
+let { NickelDerivation, Derivation, NixString, .. } = import "contracts.ncl" in
+
+let concat_strings_sep = fun sep values =>
+  if std.array.length values == 0 then "" else
+    let first = std.array.first values in
+    values
+    |> std.array.drop_first
+    |> std.array.fold_left (fun acc value => nix-s%"%{acc}%{sep}%{value}"%) first
```

Comment:
I missed that on the first review, but you can use `std.array.reduce_left` for folding starting with the first element, without having to `drop_first` manually

Reply:
https://nickel-lang.org/stdlib/std-array/#reduce_left

Reply:
Thanks, done!